src/Twig/Extension/SonataAdminExtension.php

```diff
@@ -695,4 +707,45 @@ private function getTemplateRegistry(string $adminCode): TemplateRegistryInterfa
 
         throw new ServiceNotFoundException($serviceId);
     }
+
+    /**
+     * @param object|array $listElement
+     *
+     * @return array array containing object and value
+     */
+    private function getObjectAndValueFromListElement(
+        $listElement,
+        FieldDescriptionInterface $fieldDescription
+    ): array {
+        if (\is_object($listElement)) {
+            $object = $listElement;
+        } elseif (\is_array($listElement)) {
+            if (!isset($listElement[0]) || !\is_object($listElement[0])) {
+                throw new \InvalidArgumentException('If listElement is an array it must contain an object on key 0');
```

Comment:
Similar to the next comment, I wouldn't make a sentence with `listElement` but prefer to talk about `argument 1`.

And what do you think about asking for an `non-empty` array and calling `reset()` https://www.php.net/manual/fr/function.reset.php

Reply:
I want to make absolutely sure that $listElement is the result of a Doctrine query for a model plus extra values. The shape I need in that case is an array with a value on key `0`.